src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3229 @@
+        {
+            string moduleName = "PsDesiredStateConfiguration";
+            if (!importInBoxResourcesImplicitly)
+            {
+                moduleName = Path.GetFileName(moduleFolderPath);
+            }
+
+            string manifestPath = Path.Join(moduleFolderPath, moduleName + ".psd1");
+            s_tracer.WriteLine("DSC GetModuleVersion: Try retrieving module version information from file: {0}.", manifestPath);
+
+            if (!File.Exists(manifestPath))
+            {
+                if (isPsProviderModule)
+                {
+                    // Some internal PSProviders do not come with a .psd1 file, such
+                    // as MSFT_LogResource. We don't report error in this case.
+                    return new Tuple<string, Version>(moduleName, new Version("1.0"));
+                }
+                else
+                {
+                    s_tracer.WriteLine("DSC GetModuleVersion: Manifest file '{0}' not exist.", manifestPath);
+                    return null;
+                }
+            }
+
+            try
+            {
+                Hashtable dataFileSetting =
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestModuleVersionPropertyName);
+
+                object versionValue = dataFileSetting["ModuleVersion"];
+                if (versionValue is not null)
+                {
+                    Version moduleVersion;
+                    if (LanguagePrimitives.TryConvertTo(versionValue, out moduleVersion))
+                    {
+                        return new Tuple<string, Version>(moduleName, moduleVersion);
+                    }
+                    else
+                    {
+                        s_tracer.WriteLine(
+                            "DSC GetModuleVersion: ModuleVersion value '{0}' cannot be converted to System.Version. Skip the module '{1}'.",
+                            versionValue, moduleName);
+                    }
+                }
+                else
+                {
+                    s_tracer.WriteLine(
+                        "DSC GetModuleVersion: Manifest file '{0}' does not contain ModuleVersion. Skip the module '{1}'.",
+                        manifestPath, moduleName);
+                }
+            }
+            catch (PSInvalidOperationException ex)
+            {
+                s_tracer.WriteLine(
+                    "DSC GetModuleVersion: Error evaluating module manifest file '{0}', with error '{1}'. Skip the module '{2}'.",
+                    manifestPath, ex, moduleName);
+            }
+
+            return null;
+        }
+
+
+        private static void WriteWarning(string warning)
+        {
+            var executionContext = System.Management.Automation.Runspaces.Runspace.DefaultRunspace.ExecutionContext;
```

Comment:
```suggestion
            var executionContext = Runspace.DefaultRunspace.ExecutionContext;
```

Reply:
`JsonDscClassCache.cs(437,36): error CS0103: The name 'Runspace' does not exist in the current context `